R/get_source_expressions.R

```diff
@@ -93,14 +95,25 @@ get_source_expressions <- function(filename, lines = NULL) {
       top_level_map
     )
 
+    if (!is.null(xml_parsed_content)) {
+      expression_xmls <- lapply(
+        xml2::xml_find_all(xml_parsed_content, "/exprlist/*"),
+        function(top_level_expr) xml2::xml_add_parent(xml2::xml_new_root(top_level_expr), "exprlist")
+      )
+      expressions[] <- Map(function(expr, xml) {
```

Comment:
this overwrites the whole list just to add sub-elements -- could you benchmark that vs. (maybe not exactly right) `for (ii in seq_along(expressions)) expressions[[ii]]$xml_parsed_content <- expression_xmls[[ii]]`?

Reply:
Yes, will test that.

Reply:
It is indeed 200ms faster 👍🏻

Reply:
nice! that makes this branch another 33% vs main?